src/harness/constraint_testcase.py

```diff
@@ -0,0 +1,125 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import json
+from datetime import datetime,timedelta
+import logging
+import os
+import time
+import sys
+import sas
+import sas_testcase
+from sas_test_harness import SasTestHarness, GetSasTestHarnessUrl
+
+class ConstraintTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    # Note that we disable the admin.Reset(), to avoid 'unexpected' request
+    # to SAS UUT before a test really starts.
+    # Tests changing the SAS UUT state must explicitly call the SasReset()
+    #self.SasReset()
+    self.sasTHInstances = {}
+
+  def SasReset(self):
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+     # check if any SASTH is started and try to stop it
+     for sasTH in sorted(self.sasTHInstances.itervalues()):
+         sasTH.shutdown()
+
+  def createSASTH(self,sasTHConfig):
+     sasTHSettings = sasTHConfig['sas_hareness']
+     # prepare a HTTPServer Object for each sasTH mentioned in configuration file
+     for item in sasTHSettings:
```

Comment:
Can you pick a more descriptive name than `item`?

Reply:
replaced   the name 'item' with 'sas'